Alignment/MillePedeAlignmentAlgorithm/python/alignmentsetup/GeneralSetup.py

```diff
@@ -1,29 +1,54 @@
 from __future__ import print_function
-def setup(process, global_tag, zero_tesla = False):
+import re
+
+def setup(process, global_tag, zero_tesla=False, geometry=""):
     """General setup of an alignment process.
 
     Arguments:
     - `process`: cms.Process object
     - `global_tag`: global tag to be used
     - `zero_tesla`: if 'True' the B-field map for 0T is enforced
+    - `geometry`: geometry to be used (default is an empty string for the standard geometry)
     """
 
     # MessageLogger for convenient output
     # --------------------------------------------------------------------------
     process.load('Alignment.MillePedeAlignmentAlgorithm.alignmentsetup.myMessageLogger_cff')
 
-    # Load the conditions
+    # Load the magnetic field configuration
     # --------------------------------------------------------------------------
     if zero_tesla:
-        # actually only needed for 0T MC samples, but does not harm for 0T data:
+        # For 0T MC samples or data
         process.load("Configuration.StandardSequences.MagneticField_0T_cff")
     else:
         process.load('Configuration.StandardSequences.MagneticField_cff')
-    process.load('Configuration.Geometry.GeometryRecoDB_cff')
+
+    # Load the geometry
+    # --------------------------------------------------------------------------
+    if geometry == "":
+        # Default geometry
+        print(f"Using Geometry from DB")
+        process.load('Configuration.Geometry.GeometryRecoDB_cff')
+    else:
+        # Check if the geometry string matches the format "Extended<X>", e.g. Extended2026D110
```

Comment:
`Extended2026D110` has become by now `ExtendedRun4D110`, but ok: this is just an example in the comment field that explains the usage, not meant ton e correspond exactly to an existing geometry